adapter/authzOpa/authzOpa.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2017 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package authzOpa // import "istio.io/mixer/adapter/authzOpa"
+
+// NOTE: This adapter will eventually be auto-generated so that it automatically supports all CHECK and QUOTA
+//       templates known to Mixer. For now, it's manually curated.
+
+import (
+	"context"
+	"fmt"
+
+	rpc "github.com/googleapis/googleapis/google/rpc"
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/rego"
+
+	"istio.io/mixer/adapter/authzOpa/config"
+	"istio.io/mixer/pkg/adapter"
+	"istio.io/mixer/pkg/status"
+	"istio.io/mixer/template/authz"
+)
+
+type (
+	builder struct {
+		adapterConfig *config.Params
+	}
+
+	handler struct {
+		policy      string
+		checkMethod string
+
+		context  context.Context
```

Comment:
Sorry, missed this on the first pass: why do you hold on to the context here? I don't think you need it (you don't use it AFAIK).

Reply:
Removed